src/Cli/Microsoft.DotNet.Configurer/DotnetFirstTimeUseConfigurer.cs

```diff
@@ -41,152 +39,83 @@ public class DotnetFirstTimeUseConfigurer
             _pathAdder = pathAdder ?? throw new ArgumentNullException(nameof(pathAdder));
             _performanceMeasurements ??= performanceMeasurements;
         }
 
         public void Configure()
         {
-            if (ShouldAddPackageExecutablePath())
+            if (_dotnetFirstRunConfiguration.AddGlobalToolsToPath && !_toolPathSentinel.Exists())
             {
-                Stopwatch beforeAddPackageExecutablePath = Stopwatch.StartNew();
-                AddPackageExecutablePath();
-                _performanceMeasurements?.Add("AddPackageExecutablePath Time", beforeAddPackageExecutablePath.Elapsed.TotalMilliseconds);
+                using (new PerformanceMeasurement(_performanceMeasurements, "AddPackageExecutablePath Time"))
+                {
+                    _pathAdder.AddPackageExecutablePathToUserPath();
+                    _toolPathSentinel.Create();
+                }
             }
 
-            var isFirstTimeUse = ShouldPrintFirstTimeUseNotice();
+            var isFirstTimeUse = !_firstTimeUseNoticeSentinel.Exists();
             var canShowFirstUseMessages = isFirstTimeUse && !_dotnetFirstRunConfiguration.NoLogo;
             if (isFirstTimeUse)
             {
-                Stopwatch beforeFirstTimeUseNotice = Stopwatch.StartNew();
-                // Migrate the nuget state from earlier SDKs
-                NuGet.Common.Migrations.MigrationRunner.Run();
-
-                if (canShowFirstUseMessages)
+                using (new PerformanceMeasurement(_performanceMeasurements, "FirstTimeUseNotice Time"))
                 {
-                    PrintFirstTimeMessageWelcome();
-                    if (ShouldPrintTelemetryMessageWhenFirstTimeUseNoticeIsEnabled())
+                    // Migrate the NuGet state from earlier SDKs
+                    NuGet.Common.Migrations.MigrationRunner.Run();
+
+                    if (canShowFirstUseMessages)
                     {
-                        PrintTelemetryMessage();
+                        _reporter.WriteLine();
+                        string productVersion = Product.Version;
+                        _reporter.WriteLine(string.Format(LocalizableStrings.FirstTimeMessageWelcome, ParseDotNetVersion(productVersion), productVersion));
+
+                        if (!_dotnetFirstRunConfiguration.TelemetryOptout)
+                        {
+                            _reporter.WriteLine();
+                            _reporter.WriteLine(LocalizableStrings.TelemetryMessage);
+                        }
                     }
-                }
 
-                _firstTimeUseNoticeSentinel.CreateIfNotExists();
-                _performanceMeasurements?.Add("FirstTimeUseNotice Time", beforeFirstTimeUseNotice.Elapsed.TotalMilliseconds);
+                    _firstTimeUseNoticeSentinel.CreateIfNotExists();
+                }
             }
 
-            if (ShouldGenerateAspNetCertificate())
+            if (CanGenAspNetCert())
```

Comment:
Nit: CanGenerateAspNETCertificate (I prefer the written out version personally and we've done that most everywhere else in this file as far as I can tell)

Reply:
I've updated this method name. Good call.